### PR TITLE
Feat rope: enable npu_mrope by environment variables

### PR DESCRIPTION
**Pull request by Ronald1995**
### What this PR does / why we need it?
npu_rotary_embedding will trigger gatherv3's error, enable npu_mrope to solve this issue.

### Does this PR introduce _any_ user-facing change?
export VLLM_ASCEND_ENABLE_NPU_MROPE=1 to enable npu_mrope

### How was this patch tested?
use ut to test this patch

### CODE REVIEW

**Comment by jianzs** [comment]
Thank you for your PR. A brief and clear description of your changes would be helpful.

File: vllm_ascend/envs.py [hunk 1]
```diff
@@ -133,6 +133,12 @@
     # value to False to disable the optimized model.
     "USE_OPTIMIZED_MODEL":
     lambda: bool(int(os.getenv('USE_OPTIMIZED_MODEL', '1'))),
+    # VLLM_ASCEND_ENABLE_NPU_MROPE:
+    #   0: using npu_rotary_embedding.
+    #   1: using npu_mrope.
+    # Just a temporary plan，will be removed after npu_mrope supports aclgraph mode.
+    "VLLM_ASCEND_ENABLE_NPU_MROPE":
```
Comment (Yikun) [inline vllm_ascend/envs.py:140]: Why we need this ENV? Can we use the `npu_mrope` as default?

IMO, either it is stable enough that we merge it and set it as default. Or it is not stable enough and we are not ready to merge it yet.

cc @ganyi1996ppo

File: vllm_ascend/ops/rotary_embedding.py [hunk 2]
```diff
@@ -20,13 +20,20 @@
 
 import torch
 from vllm.model_executor.layers.rotary_embedding import (
-    DeepseekScalingRotaryEmbedding, RotaryEmbedding)
+    DeepseekScalingRotaryEmbedding,
+    RotaryEmbedding,
+)
```
Comment (Yikun) [inline vllm_ascend/ops/rotary_embedding.py]: Could we revert all unrelated lint changes? because this will make cherry pick as a diffcult thing.

File: vllm_ascend/envs.py (hunk 1 above)
Comment (wangxiyuan) [reply vllm_ascend/envs.py:140]: npu_mrope doesn't work with torchair.

Comment (ganyi1996ppo) [reply vllm_ascend/envs.py:140]: If the `npu_rotary_embedding` have functionality issue, we should push torch_npu or the kernel-dev group to fix this bug rather than use a new op to replace the old one, right?

Comment (ganyi1996ppo) [reply vllm_ascend/envs.py:140]: And it there any performance assessment on this kernel?

File: vllm_ascend/ops/rotary_embedding.py (hunk 2 above)
Comment (Ronald1995) [reply vllm_ascend/ops/rotary_embedding.py]: ok，i have reverted unrelated lint changes

**Comment by wangxiyuan** [comment]
For reviewer: this pr replace `npu_rotary_embedding ` to  `npu_mrope` to solve gatherV3 bug. While a new problem is that `npu_mrope` doesn't work with aclgraph currently. So a new ENV is added in this PR. it's guaranteed that the error will be fixed in the next pta release. Then the ENV  can be removed. So I'm fine with this change for quick fix case.

**Comment by wangxiyuan** [comment]
@Ronald1995  Just like @jianzs comment, please fill up your commit message. It should include the below info at least: 
1. why the `npu_rotary_embedding ` doesn't work.
2.  what case you fixed with PR.
3. what will be done in the next step

File: vllm_ascend/envs.py (hunk 1 above)
Comment (Ronald1995) [reply vllm_ascend/envs.py:140]: this pr replace npu_rotary_embedding  to npu_mrope to solve gatherV3 bug. While a new problem is that npu_mrope doesn't work with aclgraph currently. So a new ENV is added in this PR. it's guaranteed that the error will be fixed in the next pta release. Then the ENV can be removed.

**Comment by Yikun** [comment]
Wait for @ganyi1996ppo decision, have a offline discussion with @ganyi1996ppo he had a simple way to resolve gather v3 error

**Comment by Yikun** [comment]
> this pr replace npu_rotary_embedding to npu_mrope to solve gatherV3 bug. 

This is a wrong info.

Does this PR still needed?

I think if we think it's a good way to enhancement accuracy for RL case, we could introduce a ENV for RL, rather than separate standalone var for every features.

**Comment by Yikun** [comment]
looks like it's no needed any more. feel free to reopen if you still need this.
